background/main.ts

```diff
@@ -549,10 +536,16 @@ export default class Main extends BaseService<never> {
       serializer: encodeJSON,
       deserializer: decodeJSON,
       diffStrategy: (oldObj, newObj, forceUpdate) => {
-        storeUpdateLock = new Promise((resolve) => {
-          pendingUpdates.push(resolve)
-          queueUpdate(oldObj, newObj, forceUpdate)
-        })
+        if (!storeUpdateLock) {
+          storeUpdateLock = new Promise((resolve) => {
+            releaseLock = () => {
+              resolve()
+              storeUpdateLock = null
+            }
+          })
+        }
```

Comment:
Golfing at this point, but:

```suggestion
        storeUpdateLock ??= new Promise((resolve) => {
          releaseLock = () => {
            resolve()
            storeUpdateLock = null
          }
        }
```